src/main/java/com/fundynamic/d2tm/game/math/BoundedVector2D.java

```diff
@@ -0,0 +1,33 @@
+package com.fundynamic.d2tm.game.math;
+
+
+public class BoundedVector2D<T extends Number> extends Vector2D {
```

Comment:
I having trouble distinguishing the difference between a BoundedVector2D and a simple Rect. How is this useful?